hoomd/hpmc/integrate_nec.py

```diff
@@ -0,0 +1,300 @@
+                                        param_dict=TypeParameterDict(
+                                            diameter=float,
+                                            ignore_statistics=False,
+                                            orientable=False,
+                                            len_keys=1))
+        self._add_typeparam(typeparam_shape)
+
+    @log(category='object')
+    def type_shapes(self):
+        """list[dict]: Description of shapes in ``type_shapes`` format.
+
+        Examples:
+            The types will be 'Sphere' regardless of dimensionality.
+
+            >>> mc.type_shapes
+            [{'type': 'Sphere', 'diameter': 1},
+             {'type': 'Sphere', 'diameter': 2}]
+        """
+        return super()._return_type_shapes()
+
+
+class ConvexPolyhedron(HPMCNECIntegrator):
+    """HPMC integration for convex polyhedra (3D) with nec.
+
+    Args:
+        d (float): Maximum move displacement, Scalar to set for all types,
+            or a dict containing {type:size} to set by type.
+        a (float): Maximum rotation move, Scalar to set for all types, or a
+            dict containing {type:size} to set by type.
+        chain_probability (float): Ratio of chains to rotation moves. As
+            there should be several particles in a chain it will be small.
+            See example.
+        chain_time (float):
+        update_fraction (float):
+        nselect (int): Number of repeated updates for the cell/system.
+
+    Convex polyhedron parameters:
+        see ``ConvexPolyhedron``
+
+    Warning:
+        HPMC does not check that all requirements are met. Undefined behavior
+        will result if they are violated.
+
+    Example:
+        cpu = hoomd.device.CPU()
+        sim = hoomd.Simulation(device=cpu)
+        sim.create_state_from_gsd(filename='start.gsd')
+
+        sim.state.thermalize_particle_momenta(hoomd.filter.All(), kT=1)
+
+        mc = hoomd.hpmc.integrate_nec.ConvexPolyhedron(d=1.0, a=0.05,
+            chain_probability=0.1, nselect=10)
+        mc.shape['A'] = dict(vertices=[[1,1,1], [1,1,-1], [1,-1,1], [1,-1,-1],
+            [-1,1,1], [-1,1,-1], [-1,-1,1], [-1,-1,-1]])
+        mc.chain_time = 0.05
+        sim.operations.integrator = mc
+
+        triggerTune = hoomd.trigger.Periodic(50,0)
+        tune_nec_d = hoomd.hpmc.tune.MoveSize.scale_solver(triggerTune,
+                        moves=['d'], target=0.10, max_translation_move=0.15)
+        sim.operations.tuners.append(tune_nec_d)
+        tune_nec_a = hoomd.hpmc.tune.MoveSize.scale_solver(triggerTune,
+                        moves=['a'], target=0.30)
+        sim.operations.tuners.append(tune_nec_a)
+
+        import hoomd.hpmc.tune.nec_chain_time
+        tune_nec_ct = hoomd.hpmc.tune.ChainTime.scale_solver(
+                        triggerTune, target=20, tol=1, gamma=20 )
+        sim.operations.tuners.append(tune_nec_ct)
+
+        sim.run(1000)
```

Comment:
We should probably use a minimal example here rather than a complete working simulation (assuming start.gsd exists). @joaander what do you think of long examples like this?

This applies to the class above too.

Reply:
I think even minimal examples in the API docs are prone to syntax errors and will likely be taken out of context.